src/query/string_opfunc.c

```diff
@@ -3254,6 +3254,208 @@ db_json_insert (DB_VALUE * result, DB_VALUE * arg[], int const num_args)
+{
+  int i, error_code = NO_ERROR;
+  JSON_DOC *new_doc = NULL;
+  char *str;
+
+  if (num_args < 3)
+    {
+      return DB_MAKE_NULL (result);
+    }
+
+  if (DB_IS_NULL (arg[0]))
+    {
+      return DB_MAKE_NULL (result);
+    }
+
+  switch (DB_VALUE_DOMAIN_TYPE (arg[0]))
+    {
+    case DB_TYPE_CHAR:
+    case DB_TYPE_VARCHAR:
+    case DB_TYPE_NCHAR:
+    case DB_TYPE_VARNCHAR:
+      error_code = db_json_get_json_from_str (DB_PULL_STRING (arg[0]), new_doc);
+      if (error_code != NO_ERROR)
+	{
+	  assert (new_doc == NULL);
+	  return error_code;
+	}
+      break;
+
+    case DB_TYPE_JSON:
+      new_doc = db_json_get_copy_of_doc (arg[0]->data.json.document);
+      break;
+    }
+
+  for (i = 1; i < num_args; i += 2)
+    {
+      if (DB_IS_NULL (arg[i]))
+	{
+	  db_json_delete_doc (new_doc);
+	  return DB_MAKE_NULL (result);
+	}
+
+      switch (DB_VALUE_DOMAIN_TYPE (arg[i + 1]))
+	{
+	case DB_TYPE_CHAR:
+	  error_code = db_json_convert_string_and_call (DB_PULL_STRING (arg[i + 1]),
+							db_json_replace_func, new_doc, DB_PULL_STRING (arg[i]));
+	  break;
+
+	case DB_TYPE_JSON:
+	  error_code = db_json_replace_func (arg[i + 1]->data.json.document, new_doc, DB_PULL_STRING (arg[i]));
+	  break;
+	case DB_TYPE_NULL:
+	  db_json_delete_doc (new_doc);
+	  return DB_MAKE_NULL (result);
+	default:
+	  db_json_delete_doc (new_doc);
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_QSTR_INVALID_DATA_TYPE, 0);
+	  return ER_QSTR_INVALID_DATA_TYPE;
+	}
+
+      if (error_code != NO_ERROR)
+	{
+	  db_json_delete_doc (new_doc);
+	  return error_code;
+	}
+    }
+
+  str = db_json_get_raw_json_body_from_document (new_doc);
+  db_make_json (result, str, new_doc, true);
+
+  return NO_ERROR;
+}
+
+int
+db_json_set (DB_VALUE * result, DB_VALUE * arg[], int const num_args)
+{
+  int i, error_code = NO_ERROR;
+  JSON_DOC *new_doc = NULL;
+  char *str;
+
+  if (num_args < 3)
+    {
+      return DB_MAKE_NULL (result);
+    }
+
+  if (DB_IS_NULL (arg[0]))
+    {
+      return DB_MAKE_NULL (result);
+    }
+
+  switch (DB_VALUE_DOMAIN_TYPE (arg[0]))
+    {
+    case DB_TYPE_CHAR:
+      error_code = db_json_get_json_from_str (DB_PULL_STRING (arg[0]), new_doc);
+      if (error_code != NO_ERROR)
+	{
+	  assert (new_doc == NULL);
+	  return error_code;
+	}
+      break;
+
+    case DB_TYPE_JSON:
+      new_doc = db_json_get_copy_of_doc (arg[0]->data.json.document);
+      break;
+    }
+
+  for (i = 1; i < num_args; i += 2)
+    {
+      if (DB_IS_NULL (arg[i]))
+	{
+	  db_json_delete_doc (new_doc);
+	  return DB_MAKE_NULL (result);
+	}
+
+      switch (DB_VALUE_DOMAIN_TYPE (arg[i + 1]))
+	{
+	case DB_TYPE_CHAR:
+	  error_code = db_json_convert_string_and_call (DB_PULL_STRING (arg[i + 1]),
+							db_json_set_func, new_doc, DB_PULL_STRING (arg[i]));
+	  break;
+
+	case DB_TYPE_JSON:
+	  error_code = db_json_set_func (arg[i + 1]->data.json.document, new_doc, DB_PULL_STRING (arg[i]));
+	  break;
+	case DB_TYPE_NULL:
+	  db_json_delete_doc (new_doc);
```

Comment:
Confirm this is indented behavior: if any argument is NULL, then the result is also NULL.